Added new "Plugin metadata" section in docs (#9529)
ckeditor/ckeditor5: docs/framework/guides/contributing/package-metadata.md

## Data format for plugin metadata

The `ckeditor5-metadata.json` file is a JSON object that holds the `plugins` array. Each element of this array should be an object containing information about a different plugin delivered by the package. Below is a list of all matadata properties that can be used to describe the plugin:

* `name` &ndash; A human-readable name of the plugin.
* `description` &ndash; A human-readable short description of the plugin.
* `docs` &ndash; An absolute or relative URL for the plugin's documentation. If this URL is relative, it leads to the CKEditor 5 documentation in [https://ckeditor.com/docs/ckeditor5/latest/](https://ckeditor.com/docs/ckeditor5/latest/).
* `path` &ndash; A path to the file, relative to the metadata file that exports the plugin.

> * `className` &ndash; The name of the class used to create the plugin.

Reviewer: \+ This class should be exported from the file using the `export default` syntax.

Author: Done.

> * `requires` &ndash; An array of the plugin's soft requirements and other non-explicit requirements.

Reviewer: \+ It should contain class names of plugins that should be included if this plugin is added.

Reviewer: Perhaps it would be safer to add support for `repository:pluginName` in the future.

Author: Done.

> * `registeredToolbars` &ndash; An array of all toolbar names registered by the plugin. These names need to contain the full configuration path (e.g. `table.contentToolbar` and `table.tableToolbar` for `Table` plugin).

Reviewer: ```suggestion
* `registeredToolbars` &ndash; An array of all toolbar names registered by the plugin. These names need to represent the configuration path (e.g. `table.contentToolbar` for `editorConfig.table.contentToolbar` and `table.tableToolbar` for the `editorConfig.tableToolbar` are registered by the `Table` plugin).
```

Author: Done.
The `tableToolbar` is also inside `editorConfig.table.tableToolbar`.

> * `uiComponents` &ndash; An array of objects, that describe UI components exported by the plugin. Each object in this array may contain:

Reviewer: ```suggestion
* `uiComponents` &ndash; An array of objects, that describes UI components exported by the plugin. Each object in this array may contain:
```

Author: Done.

> 	* `name` &ndash; A name of the component the plugin exports.

Reviewer: It should match the actual UI name registered by the plugin.

Author: Done.

	* `type` &ndash; Component type: `Button`, `SplitButton` or `Dropdown`.
	* `iconPath` &ndash; A path to the SVG icon for `Button` or `SplitButton` components.
	* `label` &ndash; Text content for `Dropdown` components.

> 	* `toolbars` &ndash; An array of toolbar names, a given UI component can be added to. Some features may be added to multiple toolbars.

Reviewer: ```suggestion
	* `toolbars` &ndash; An array of toolbar names, a given UI component can be added to. Some UI componenets may be added to multiple toolbars.
```

Author: Done.

> * `htmlOutput` &ndash; An array of objects, that define all possible HTML elements, which can be created by a given plugin. The main property in this object is `elements`. Other properties (e.g. `classes`, `styles`, `attributes`) only apply to items defined in the `elements` property within a given object. Wildcard character `*` is used to mark any value. Full list of all these properties includes:

Reviewer: ```suggestion
* `htmlOutput` &ndash; An array of objects, that defines all possible HTML elements, which can be created by a given plugin. The main property in this object is `elements`. Other properties (e.g. `classes`, `styles`, `attributes`) only apply to items defined in the `elements` property within a given object. Wildcard character `*` is used to mark any value. Full list of all these properties includes:
```

Author: Done.

> 	* `elements` &ndash; HTML elements (a single one or an array of these) that are created by the plugin. The pseudo-element `$block` indicates that a given plugin applies classes, styles or attributes (defined in appropriate properties) for all block elements.

Reviewer: I'd rephrase it as the first sentence is incorrect for the \`$block\`

Author: Changed "(...) that are created" to "(...) that are created or altered".

* `htmlOutput` &ndash; An array of objects, that defines all possible HTML elements, which can be created by a given plugin. The main property in this object is `elements`. Other properties (e.g. `classes`, `styles`, `attributes`) only apply to items defined in the `elements` property within a given object. Wildcard character `*` is used to mark any value. Full list of all these properties includes:
	* `elements` &ndash; HTML elements (a single one or an array of these) that are created or altered by the plugin. The pseudo-element `$block` indicates that a given plugin applies classes, styles or attributes (defined in appropriate properties) for all block elements.
	* `classes` &ndash; CSS class names (a single one or an array of these) that may be applied to the HTML elements defined in the `elements` property.
	* `styles` &ndash; Inline CSS styles (a single one or an array of these) that may be applied to the HTML elements defined in the `elements` property.
	* `attributes` &ndash; HTML attributes (a single one or an array of these) other than `class` and `styles` (covered separately), that might be applied to the HTML elements defined in the `elements` property.

> 	* `implements` &ndash; A name of an element or a pseudo-element which indicates that HTML elements defined in the `elements` property may contain classes, styles or attributes that are created by other plugins. For example, `implements` equal to `$block` means, that HTML elements may contain classes, styles or attributes that are defined by another plugin, which have `elements` equal to `$block`.

Reviewer 2: ```suggestion
> For example, `implements` equal to `$block` means, that HTML elements may contain classes, styles or attributes that are defined by another plugin, which have `elements` equal to `$block`.
	
DO the classe, styles or attributes have `element`, or DOES the plugin have these? It is not clear now and if the latter, there needs to be 

> which has `elements` (...)	
```

> 	* `implements` &ndash; A name of an element or a pseudo-element which indicates that HTML elements defined in the `elements` property may contain classes, styles or attributes that are created by other plugins. For example, `implements` equal to `$block` means, that HTML elements may contain classes, styles or attributes that are defined by another plugin, which have `elements` equal to `$block`.

Reviewer 2: Here:  
  
\> classes, styles or attributes that are defined by another plugin, which have (...)  
  
if it's about the classes etc. or  
  
\> classes, styles or attributes that are defined by another plugin, which has (...)  
  
if it's about the plugin. This is not exactly clear.

Author: It is about the plugin, so I've just changed it to _"classes, styles or attributes that are defined by another plugin, which **has** (...)"_. Thanks.

Reviewer 2: Cool, thanks!
